blosc/blosc.c

```diff
@@ -1422,6 +1422,63 @@ int blosc_compress(int clevel, int doshuffle, size_t typesize, size_t nbytes,
   return result;
 }
 
+static size_t blosc_compression_bound_helper(size_t cursize, int compressor)
+{
+  switch (compressor)
+  {
+  case BLOSC_BLOSCLZ:
+    // from doc : The output buffer must be at least 5% larger than the input buffer
+    // and can not be smaller than 66 bytes.
+    return max(66, (size_t)ceil(1.05 * cursize));
+
+#if defined(HAVE_LZ4)
+  case BLOSC_LZ4:
+  case BLOSC_LZ4HC:
+    return (size_t)LZ4_compressBound((int)cursize);
+#endif /* HAVE_LZ4 */
+
+#if defined(HAVE_SNAPPY)
+  case BLOSC_SNAPPY:
+    return snappy_max_compressed_length(cursize);
+#endif /* HAVE_SNAPPY */
+
+#if defined(HAVE_ZLIB)
+  case BLOSC_ZLIB:
+    return (size_t)compressBound((uLong)cursize);
+#endif /* HAVE_ZLIB */
+
+#if defined(HAVE_ZSTD)
+  case BLOSC_ZSTD:
+    return ZSTD_compressBound(cursize);
+#endif /* HAVE_ZSTD */
+
+  default:
+  {
+    const char *compname;
+    compname = clibcode_to_clibname(compressor);
+    if (compname == NULL) {
+        compname = "(null)";
+    }
+    fprintf(stderr, "Blosc has not been compiled with '%s' ", compname);
+    fprintf(stderr, "compression support.  Please use one having it.");
+    return 0;    /* failure */
+    break;
+  }
+  }
+}
+
+size_t blosc_compression_bound_context(size_t cursize, struct blosc_context* context)
```

Comment:
Please include this function in the public header file as well, and duplicate the documentation from `blosc_compression_bound()`. After that, at least I'm happy with the changes and we will have to wait for someone with commit rights :)

Reply:
Great, thank you for your responsiveness

Reply:
Actually, I don't think I can. That struct is private

Reply:
You are right. Looking at other _ctx functions, it seems all necessary information is passed as arguments instead of having a context pointer. In this case it would be (I don't know if blocksize is needed):

  `compressor`: the string representing the type of compressor to use.
  `blocksize`: the requested size of the compressed blocks.  If 0, an
   automatic blocksize will be used.

Reply:
In general, I'd say that the function can be much simplified using just:

```
static size_t blosc_compression_bound_helper(size_t cursize) {
 return BLOSC_MAX_OVERHEAD + cursize;
}
```

I don't know if having such a function buys a lot for the user though.  Although if it mimics existing function in zlib, maybe it saves some time for users looking for this.

Reply:
Do you think so? Each underlying compressor requires a different amount of size

Reply:
Right, but there is an internal check controlling the final size of the compressed buffer.  If it is larger than `BLOSC_MAX_OVERHEAD + nbytes`, the compression stops and a literal copy is returned instead (plus the header).

Reply:
I see. Then this function loses much usefulness. A simple documentation change would be enough

Reply:
I'd think that the existing doc in https://github.com/Blosc/c-blosc/blob/master/blosc/blosc.h#L163 should be enough (but feel free to open a new PR for amending this differently).